# Project - Shark Attacks v2

## Libraries

In [1]:
import pandas as pd
import numpy as np

# EDA

## Importing dataset

In [2]:
attacks = pd.read_csv('attacks.csv', encoding='latin1')
pd.options.display.max_columns = None
pd.options.display.max_rows = None
attacks.head(5)

Case Number         Date    Year        Type    Country             Area  \
0  2018.06.25  25-Jun-2018  2018.0     Boating        USA       California   
1  2018.06.18  18-Jun-2018  2018.0  Unprovoked        USA          Georgia   
2  2018.06.09  09-Jun-2018  2018.0     Invalid        USA           Hawaii   
3  2018.06.08  08-Jun-2018  2018.0  Unprovoked  AUSTRALIA  New South Wales   
4  2018.06.04  04-Jun-2018  2018.0    Provoked     MEXICO           Colima   

                         Location     Activity             Name Sex   Age  \
0     Oceanside, San Diego County     Paddling      Julie Wolfe    F   57   
1  St. Simon Island, Glynn County     Standing  Adyson McNeely     F   11   
2                    Habush, Oahu      Surfing      John Denges    M   48   
3              Arrawarra Headland      Surfing             male    M  NaN   
4                        La Ticla  Free diving   Gustavo Ramos     M  NaN   

                                              Injury Fatal (Y/N)  \
0  No injury to occupant, outrigger canoe and pad...           N   
1                         Minor injury to left thigh           N   
2       Injury to left lower leg from surfboard skeg           N   
3                          Minor injury to lower leg           N   
4  Lacerations to leg & hand shark PROVOKED INCIDENT           N   

            Time         Species           Investigator or Source  \
0          18h00      White shark                R. Collier, GSAF   
1  14h00  -15h00              NaN  K.McMurray, TrackingSharks.com   
2          07h45              NaN  K.McMurray, TrackingSharks.com   
3            NaN        2 m shark                  B. Myatt, GSAF   
4            NaN  Tiger shark, 3m                       A .Kipper   

                        pdf  \
0      2018.06.25-Wolfe.pdf   
1    2018.06.18-McNeely.pdf   
2     2018.06.09-Denges.pdf   
3  2018.06.08-Arrawarra.pdf   
4      2018.06.04-Ramos.pdf   

                                        href formula  \
0  http://sharkattackfile.net/spreadsheets/pdf_di...   
1  http://sharkattackfile.net/spreadsheets/pdf_di...   
2  http://sharkattackfile.net/spreadsheets/pdf_di...   
3  http://sharkattackfile.net/spreadsheets/pdf_di...   
4  http://sharkattackfile.net/spreadsheets/pdf_di...   

                                                href Case Number.1  \
0  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.25   
1  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.18   
2  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.09   
3  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.08   
4  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.04   

  Case Number.2  original order Unnamed: 22 Unnamed: 23  
0    2018.06.25          6303.0         NaN         NaN  
1    2018.06.18          6302.0         NaN         NaN  
2    2018.06.09          6301.0         NaN         NaN  
3    2018.06.08          6300.0         NaN         NaN  
4    2018.06.04          6299.0         NaN         NaN

## Checking for countries

In [3]:
attacks.Country.unique()

array(['USA', 'AUSTRALIA', 'MEXICO', 'BRAZIL', 'ENGLAND', 'SOUTH AFRICA',
       'THAILAND', 'COSTA RICA', 'MALDIVES', 'BAHAMAS', 'NEW CALEDONIA',
       'ECUADOR', 'MALAYSIA', 'LIBYA', nan, 'CUBA', 'MAURITIUS',
       'NEW ZEALAND', 'SPAIN', 'SAMOA', 'SOLOMON ISLANDS', 'JAPAN',
       'EGYPT', 'ST HELENA, British overseas territory', 'COMOROS',
       'REUNION', 'FRENCH POLYNESIA', 'UNITED KINGDOM',
       'UNITED ARAB EMIRATES', 'PHILIPPINES', 'INDONESIA', 'CHINA',
       'COLUMBIA', 'CAPE VERDE', 'Fiji', 'DOMINICAN REPUBLIC',
       'CAYMAN ISLANDS', 'ARUBA', 'MOZAMBIQUE', 'FIJI', 'PUERTO RICO',
       'ITALY', 'ATLANTIC OCEAN', 'GREECE', 'ST. MARTIN', 'FRANCE',
       'PAPUA NEW GUINEA', 'TRINIDAD & TOBAGO', 'KIRIBATI', 'ISRAEL',
       'DIEGO GARCIA', 'TAIWAN', 'JAMAICA', 'PALESTINIAN TERRITORIES',
       'GUAM', 'SEYCHELLES', 'BELIZE', 'NIGERIA', 'TONGA', 'SCOTLAND',
       'CANADA', 'CROATIA', 'SAUDI ARABIA', 'CHILE', 'ANTIGUA', 'KENYA',
       'RUSSIA', 'TURKS & CAICOS', 'UNITE

In [4]:
attacks_by_country = attacks.groupby(by=['Country']).count()['Case Number']
#pd.DataFrame(attacks_by_country)
attacks_by_country = pd.DataFrame(attacks_by_country.sort_values(ascending=False))
attacks_by_country

Case Number
Country                                           
USA                                           2228
AUSTRALIA                                     1338
SOUTH AFRICA                                   579
PAPUA NEW GUINEA                               134
NEW ZEALAND                                    128
BRAZIL                                         112
BAHAMAS                                        109
MEXICO                                          89
ITALY                                           71
FIJI                                            62
PHILIPPINES                                     61
REUNION                                         60
NEW CALEDONIA                                   53
CUBA                                            46
MOZAMBIQUE                                      45
SPAIN                                           44
INDIA                                           40
EGYPT                                           38
JAPAN                                           34
CROATIA                                         34
PANAMA                                          32
SOLOMON ISLANDS                                 30
IRAN                                            29
JAMAICA                                         27
GREECE                                          25
FRENCH POLYNESIA                                25
HONG KONG                                       24
ENGLAND                                         23
INDONESIA                                       23
ATLANTIC OCEAN                                  17
COSTA RICA                                      17
PACIFIC OCEAN                                   17
BERMUDA                                         16
VIETNAM                                         15
TONGA                                           15
SRI LANKA                                       14
VANUATU                                         14
FRANCE                                          13
MARSHALL ISLANDS                                13
SOUTH ATLANTIC OCEAN                            12
TURKEY                                          12
IRAQ                                            12
CANADA                                          12
UNITED KINGDOM                                  11
VENEZUELA                                       11
SENEGAL                                         11
KENYA                                           10
NEW GUINEA                                      10
MAURITIUS                                       10
COLUMBIA                                         9
TAIWAN                                           9
ECUADOR                                          9
THAILAND                                         8
CARIBBEAN SEA                                    8
TANZANIA                                         8
SAMOA                                            8
SOUTH KOREA                                      8
SCOTLAND                                         8
SIERRA LEONE                                     8
CHILE                                            8
MADAGASCAR                                       8
INDIAN OCEAN                                     7
DOMINICAN REPUBLIC                               7
ISRAEL                                           7
CHINA                                            7
YEMEN                                            7
NORTH PACIFIC OCEAN                              7
SEYCHELLES                                       7
LIBYA                                            6
NEW BRITAIN                                      6
OKINAWA                                          6
BARBADOS                                         6
NICARAGUA                                        6
KIRIBATI                                         6
SINGAPORE                                        6
SOMALIA                                          6
MALAYSIA                                         5
AZORES    

In [5]:
attacks_by_country.loc[attacks_by_country.index.isin(['BRAZIL'])]

Case Number
Country             
BRAZIL           112

BRAZIL has 112 cases overall, let us investigate it

# Exploring Cases in Brazil

In [6]:
attacks_br = attacks[attacks.Country.isin(['BRAZIL'])]
attacks_br.head()

Case Number                  Date    Year        Type Country  \
6   2018.06.03.a           03-Jun-2018  2018.0  Unprovoked  BRAZIL   
27  2018.04.15.c           15-Apr-2018  2018.0  Unprovoked  BRAZIL   
31  2018.04.10.R  Reported 10-Apr-2018  2018.0     Invalid  BRAZIL   
44    2018.02.11           11-Feb-2018  2018.0     Invalid  BRAZIL   
51    2018.01.12           12-Jan-2018  2018.0  Unprovoked  BRAZIL   

                    Area                   Location       Activity  \
6             Pernambuco      Piedade Beach, Recife       Swimming   
27            Pernambuco                    Piedade       Swimming   
31               Alagoas  Praia de Sauaçuhy, Maceió        Fishing   
44  Boi Island, Victoria             Espirito Santo  Cleaning fish   
51   Fernando de Noronha            Conceicao Beach        Surfing   

                        Name Sex  Age  \
6     Jose Ernesto da Silva     M  18   
27             Pablo de Melo    M  34   
31                Josias Paz    M  56   
44            Rosalida Souza    F  46   
51  Ricardo Ferrari Bulhoes     M  20   

                                               Injury Fatal (Y/N)  \
6                                               FATAL           Y   
27  Multiple severe injuries to arms and leg, leg ...         NaN   
31  Injury to ankle from marine animal trapped in ...           N   
44                Lacerations to 4 toes of right foot           N   
51                       Minor injury to left forearm           N   

              Time                                       Species   \
6   Late afternoon                                    Tiger shark   
27             NaN                                            NaN   
31             NaN                Shark involvement not confirmed   
44             NaN  Injury believed caused by an eel, not a shark   
51  Late afternoon                                    small shark   

             Investigator or Source                       pdf  \
6    Diario de Pernambuco, 6/4/2018  2018.06.03.a-daSilva.pdf   
27                 Globo, 4/16/2018   2018.04.15.c-deMelo.pdf   
31  K. McMurray, TrackingSharks.com      2018.04.10.R-Paz.pdf   
44               TrackingSharks.com    2018.02.11-Rosilda.pdf   
51               TrackingSharks.com    2018.01.12-Bulhoes.pdf   

                                         href formula  \
6   http://sharkattackfile.net/spreadsheets/pdf_di...   
27  http://sharkattackfile.net/spreadsheets/pdf_di...   
31  http://sharkattackfile.net/spreadsheets/pdf_di...   
44  http://sharkattackfile.net/spreadsheets/pdf_di...   
51  http://sharkattackfile.net/spreadsheets/pdf_di...   

                                                 href Case Number.1  \
6   http://sharkattackfile.net/spreadsheets/pdf_di...  2018.06.03.a   
27  http://sharkattackfile.net/spreadsheets/pdf_di...  2018.04.15.c   
31  http://sharkattackfile.net/spreadsheets/pdf_di...  2018.04.10.R   
44  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.02.11   
51  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.01.12   

   Case Number.2  original order Unnamed: 22 Unnamed: 23  
6   2018.06.03.a          6297.0         NaN         NaN  
27  2018.04.15.c          6276.0         NaN         NaN  
31  2018.04.10.R          6272.0         NaN         NaN  
44    2018.02.11          6259.0         NaN         NaN  
51    2018.01.12          6252.0         NaN         NaN

## Dropping columns with information I won't use

In [7]:
attacks_br = attacks_br.drop(['Case Number','Country','Injury','Time','Investigator or Source','pdf','href formula','href','Case Number.1','Case Number.2',
                              'original order','Unnamed: 22','Unnamed: 23'], axis=1)

In [8]:
attacks_br

Date    Year          Type                  Area  \
6              03-Jun-2018  2018.0    Unprovoked            Pernambuco   
27             15-Apr-2018  2018.0    Unprovoked            Pernambuco   
31    Reported 10-Apr-2018  2018.0       Invalid               Alagoas   
44             11-Feb-2018  2018.0       Invalid  Boi Island, Victoria   
51             12-Jan-2018  2018.0    Unprovoked   Fernando de Noronha   
177            06-Feb-2017  2017.0      Provoked            Pernambuco   
181            24-Jan-2017  2017.0    Unprovoked            Pernambuco   
302            02-Mar-2016  2016.0    Unprovoked  Santa Catarina State   
310            02-Feb-2016  2016.0    Unprovoked    Balneário Camboriú   
324            21-Dec-2015  2015.0    Unprovoked            Pernambuco   
438            31-Mar-2015  2015.0       Invalid            Pernambuco   
543            09-Jun-2014  2014.0    Unprovoked                   NaN   
651            22-Jul-2013  2013.0    Unprovoked            Pernambuco   
673            23-May-2013  2013.0    Unprovoked            Pernambuco   
751            26-Aug-2012  2012.0    Unprovoked            Pernambuco   
845            12-Nov-2011  2011.0    Unprovoked            Pernambuco   
890            31-Jul-2011  2011.0       Invalid            Pernambuco   
903            29-Jun-2011  2011.0    Unprovoked            Pernambuco   
1047           31-Jan-2010  2010.0    Unprovoked     Rio Grande Do Sul   
1090           13-Sep-2009  2009.0      Provoked            Pernambuco   
1092           07-Sep-2009  2009.0    Unprovoked            Pernambuco   
1162  Reported 26-Jan-2009  2009.0       Invalid                   NaN   
1167           23-Jan-2009  2009.0       Invalid              Maranhão   
1246           24-Jun-2008  2008.0    Unprovoked                 Bahia   
1249           11-Jun-2008  2008.0    Unprovoked            Pernambuco   
1253           01-Jun-2008  2008.0    Unprovoked            Pernambuco   
1307           07-Dec-2007  2007.0    Unprovoked            Pernambuco   
1434           03-Sep-2006  2006.0    Unprovoked                   NaN   
1441           22-Aug-2006  2006.0       Invalid            Pernambuco   
1460           10-Jul-2006  2006.0    Unprovoked            Pernambuco   
1471           18-Jun-2006  2006.0    Unprovoked            Pernambuco   
1480           21-May-2006  2006.0    Unprovoked            Pernambuco   
1491           09-Apr-2006  2006.0    Unprovoked            Pernambuco   
1613           04-Feb-2005  2005.0    Unprovoked                 Bahia   
1640           08-Sep-2004  2004.0    Unprovoked            Pernambuco   
1643           21-Aug-2004  2004.0    Unprovoked            Pernambuco   
1667           23-May-2004  2004.0    Unprovoked            Pernambuco   
1672           22-May-2004  2004.0    Unprovoked            Pernambuco   
1675           01-May-2004  2004.0    Unprovoked            Pernambuco   
1685           29-Mar-2004  2004.0    Unprovoked            Pernambuco   
1693           29-Feb-2004  2004.0    Unprovoked            Pernambuco   
1765           24-Jun-2003  2003.0    Unprovoked            Pernambuco   
1778           26-Apr-2003  2003.0      Provoked        Rio de Janeiro   
1779           25-Apr-2003  2003.0    Unprovoked        Rio de Janeiro   
1780           23-Apr-2003  2003.0    Unprovoked            Pernambuco   
1781           23-Apr-2003  2003.0    Unprovoked        Rio de Janeiro   
1807           01-Dec-2002  2002.0    Unprovoked            Pernambuco   
1816           14-Oct-2002  2002.0    Unprovoked            Pernambuco   
1828           16-Sep-2002  2002.0    Unprovoked            Pernambuco   
1844           10-Jul-2002  2002.0    Unprovoked            Pernambuco   
1863           10-May-2002  2002.0    Unprovoked            Pernambuco   
1875           24-Mar-2002  2002.0    Unprovoked            Pernambuco   
1887           03-Jan-2002  2002.0    Unprovoked            Pernambuco   
1967            02-Ap-2001

## Fillin NaN cells with 'Unknown' and removing rows with invalid information

In [9]:
attacks_br = attacks_br.fillna('Unknown')
attacks_br = attacks_br[attacks_br['Type'] != 'Invalid']
attacks_br = attacks_br[attacks_br['Type'] != 'Sea Disaster']

In [10]:
attacks_br.head()

Date    Year        Type                 Area  \
6    03-Jun-2018  2018.0  Unprovoked           Pernambuco   
27   15-Apr-2018  2018.0  Unprovoked           Pernambuco   
51   12-Jan-2018  2018.0  Unprovoked  Fernando de Noronha   
177  06-Feb-2017  2017.0    Provoked           Pernambuco   
181  24-Jan-2017  2017.0  Unprovoked           Pernambuco   

                            Location                     Activity  \
6              Piedade Beach, Recife                     Swimming   
27                           Piedade                     Swimming   
51                   Conceicao Beach                      Surfing   
177  Fernando de Noronha Marine Park  Grabbing shark for a selfie   
181               Boa Viagem, Recife                     Swimming   

                         Name Sex       Age Fatal (Y/N)         Species   
6      Jose Ernesto da Silva     M       18           Y      Tiger shark  
27              Pablo de Melo    M       34     Unknown          Unknown  
51   Ricardo Ferrari Bulhoes     M       20           N      small shark  
177                    female    F  Unknown           N  Lemon shark pup  
181            Arthur Andrade    M       21           N          Unknown

'Area' columns has information all messed up, I will replace it with States abbreviations using a DICT

In [11]:
attacks_br['Area'].unique()

array(['Pernambuco', 'Fernando de Noronha', 'Santa Catarina State',
       'Balneário Camboriú', 'Unknown', 'Rio Grande Do Sul', 'Bahia',
       'Rio de Janeiro', 'Rio Grande de Norte', 'Maranhão', 'Paraiba'],
      dtype=object)

In [12]:
myset = set(attacks_br['Area'])
keys_list = list(myset)
keys_list.sort()
values_list = ['BA','SC','PE','MA','PB','PE','RS','RN','RJ','SC','Unknown']
zip_iterator = zip(keys_list, values_list)

a_dictionary = dict(zip_iterator)

print(a_dictionary)

{'Bahia': 'BA', 'Balneário Camboriú': 'SC', 'Fernando de Noronha': 'PE', 'Maranhão': 'MA', 'Paraiba': 'PB', 'Pernambuco': 'PE', 'Rio Grande Do Sul': 'RS', 'Rio Grande de Norte': 'RN', 'Rio de Janeiro': 'RJ', 'Santa Catarina State': 'SC', 'Unknown': 'Unknown'}


In [13]:
attacks_br['State'] = attacks_br['Area'].map(a_dictionary)

In [14]:
attacks_br.head()

Date    Year        Type                 Area  \
6    03-Jun-2018  2018.0  Unprovoked           Pernambuco   
27   15-Apr-2018  2018.0  Unprovoked           Pernambuco   
51   12-Jan-2018  2018.0  Unprovoked  Fernando de Noronha   
177  06-Feb-2017  2017.0    Provoked           Pernambuco   
181  24-Jan-2017  2017.0  Unprovoked           Pernambuco   

                            Location                     Activity  \
6              Piedade Beach, Recife                     Swimming   
27                           Piedade                     Swimming   
51                   Conceicao Beach                      Surfing   
177  Fernando de Noronha Marine Park  Grabbing shark for a selfie   
181               Boa Viagem, Recife                     Swimming   

                         Name Sex       Age Fatal (Y/N)         Species  State  
6      Jose Ernesto da Silva     M       18           Y      Tiger shark    PE  
27              Pablo de Melo    M       34     Unknown          Unknown    PE  
51   Ricardo Ferrari Bulhoes     M       20           N      small shark    PE  
177                    female    F  Unknown           N  Lemon shark pup    PE  
181            Arthur Andrade    M       21           N          Unknown    PE

## Checking 'Unknown' in 'State' columns

In [15]:
attacks_br.loc[attacks_br['State'] == 'Unknown']

Date    Year        Type     Area  \
543            09-Jun-2014  2014.0  Unprovoked  Unknown   
1434           03-Sep-2006  2006.0  Unprovoked  Unknown   
5777  Reported 28-Apr-1884  1884.0  Unprovoked  Unknown   

                                Location  \
543   500 km off the coast of Pernambuco   
1434                             Unknown   
5777                               Bahia   

                                     Activity                    Name Sex   \
543                                   Fishing                 Sunarko    M   
1434                                  Unknown  Darlan dos Santos Luz     M   
5777  Fell overboard from the steamship Chala                    male    M   

          Age Fatal (Y/N) Species     State  
543        43           N  Unknown  Unknown  
1434       20           Y  Unknown  Unknown  
5777  Unknown           Y  Unknown  Unknown

## Assigning states based on location and discarding unknowns

In [16]:
attacks_br.loc[attacks_br['Location'].str.contains('Pernambuco'), 'State'] = 'PE'

In [17]:
attacks_br.loc[attacks_br['Location'].str.contains('Bahia'), 'State'] = 'BA'

In [18]:
attacks_br

Date    Year        Type                  Area  \
6              03-Jun-2018  2018.0  Unprovoked            Pernambuco   
27             15-Apr-2018  2018.0  Unprovoked            Pernambuco   
51             12-Jan-2018  2018.0  Unprovoked   Fernando de Noronha   
177            06-Feb-2017  2017.0    Provoked            Pernambuco   
181            24-Jan-2017  2017.0  Unprovoked            Pernambuco   
302            02-Mar-2016  2016.0  Unprovoked  Santa Catarina State   
310            02-Feb-2016  2016.0  Unprovoked    Balneário Camboriú   
324            21-Dec-2015  2015.0  Unprovoked            Pernambuco   
543            09-Jun-2014  2014.0  Unprovoked               Unknown   
651            22-Jul-2013  2013.0  Unprovoked            Pernambuco   
673            23-May-2013  2013.0  Unprovoked            Pernambuco   
751            26-Aug-2012  2012.0  Unprovoked            Pernambuco   
845            12-Nov-2011  2011.0  Unprovoked            Pernambuco   
903            29-Jun-2011  2011.0  Unprovoked            Pernambuco   
1047           31-Jan-2010  2010.0  Unprovoked     Rio Grande Do Sul   
1090           13-Sep-2009  2009.0    Provoked            Pernambuco   
1092           07-Sep-2009  2009.0  Unprovoked            Pernambuco   
1246           24-Jun-2008  2008.0  Unprovoked                 Bahia   
1249           11-Jun-2008  2008.0  Unprovoked            Pernambuco   
1253           01-Jun-2008  2008.0  Unprovoked            Pernambuco   
1307           07-Dec-2007  2007.0  Unprovoked            Pernambuco   
1434           03-Sep-2006  2006.0  Unprovoked               Unknown   
1460           10-Jul-2006  2006.0  Unprovoked            Pernambuco   
1471           18-Jun-2006  2006.0  Unprovoked            Pernambuco   
1480           21-May-2006  2006.0  Unprovoked            Pernambuco   
1491           09-Apr-2006  2006.0  Unprovoked            Pernambuco   
1613           04-Feb-2005  2005.0  Unprovoked                 Bahia   
1640           08-Sep-2004  2004.0  Unprovoked            Pernambuco   
1643           21-Aug-2004  2004.0  Unprovoked            Pernambuco   
1667           23-May-2004  2004.0  Unprovoked            Pernambuco   
1672           22-May-2004  2004.0  Unprovoked            Pernambuco   
1675           01-May-2004  2004.0  Unprovoked            Pernambuco   
1685           29-Mar-2004  2004.0  Unprovoked            Pernambuco   
1693           29-Feb-2004  2004.0  Unprovoked            Pernambuco   
1765           24-Jun-2003  2003.0  Unprovoked            Pernambuco   
1778           26-Apr-2003  2003.0    Provoked        Rio de Janeiro   
1779           25-Apr-2003  2003.0  Unprovoked        Rio de Janeiro   
1780           23-Apr-2003  2003.0  Unprovoked            Pernambuco   
1781           23-Apr-2003  2003.0  Unprovoked        Rio de Janeiro   
1807           01-Dec-2002  2002.0  Unprovoked            Pernambuco   
1816           14-Oct-2002  2002.0  Unprovoked            Pernambuco   
1828           16-Sep-2002  2002.0  Unprovoked            Pernambuco   
1844           10-Jul-2002  2002.0  Unprovoked            Pernambuco   
1863           10-May-2002  2002.0  Unprovoked            Pernambuco   
1875           24-Mar-2002  2002.0  Unprovoked            Pernambuco   
1887           03-Jan-2002  2002.0  Unprovoked            Pernambuco   
1967            02-Ap-2001  2001.0  Unprovoked   Rio Grande de Norte   
1972           03-Mar-2001  2001.0  Unprovoked            Pernambuco   
2082           26-Dec-1999  1999.0  Unprovoked            Pernambuco   
2090           06-Nov-1999  1999.0  Unprovoked        Rio de Janeiro   
2124           01-May-1999  1999.0  Unprovoked            Pernambuco   
2129           18-Mar-1999  1999.0    Provoked   Rio Grande de Norte   
2155           02-Nov-1998  1998.0  Unprovoked            Pernambuco   
2159           04-Oct-1998  1998.0  Unprovoked            Pernambuco   
2176           26-Jul-1998  1998.0  Unprovoked            Pern

In [19]:
attacks_br = attacks_br[attacks_br['State'] != 'Unknown']

In [20]:
attacks_br

Date    Year        Type                  Area  \
6              03-Jun-2018  2018.0  Unprovoked            Pernambuco   
27             15-Apr-2018  2018.0  Unprovoked            Pernambuco   
51             12-Jan-2018  2018.0  Unprovoked   Fernando de Noronha   
177            06-Feb-2017  2017.0    Provoked            Pernambuco   
181            24-Jan-2017  2017.0  Unprovoked            Pernambuco   
302            02-Mar-2016  2016.0  Unprovoked  Santa Catarina State   
310            02-Feb-2016  2016.0  Unprovoked    Balneário Camboriú   
324            21-Dec-2015  2015.0  Unprovoked            Pernambuco   
543            09-Jun-2014  2014.0  Unprovoked               Unknown   
651            22-Jul-2013  2013.0  Unprovoked            Pernambuco   
673            23-May-2013  2013.0  Unprovoked            Pernambuco   
751            26-Aug-2012  2012.0  Unprovoked            Pernambuco   
845            12-Nov-2011  2011.0  Unprovoked            Pernambuco   
903            29-Jun-2011  2011.0  Unprovoked            Pernambuco   
1047           31-Jan-2010  2010.0  Unprovoked     Rio Grande Do Sul   
1090           13-Sep-2009  2009.0    Provoked            Pernambuco   
1092           07-Sep-2009  2009.0  Unprovoked            Pernambuco   
1246           24-Jun-2008  2008.0  Unprovoked                 Bahia   
1249           11-Jun-2008  2008.0  Unprovoked            Pernambuco   
1253           01-Jun-2008  2008.0  Unprovoked            Pernambuco   
1307           07-Dec-2007  2007.0  Unprovoked            Pernambuco   
1460           10-Jul-2006  2006.0  Unprovoked            Pernambuco   
1471           18-Jun-2006  2006.0  Unprovoked            Pernambuco   
1480           21-May-2006  2006.0  Unprovoked            Pernambuco   
1491           09-Apr-2006  2006.0  Unprovoked            Pernambuco   
1613           04-Feb-2005  2005.0  Unprovoked                 Bahia   
1640           08-Sep-2004  2004.0  Unprovoked            Pernambuco   
1643           21-Aug-2004  2004.0  Unprovoked            Pernambuco   
1667           23-May-2004  2004.0  Unprovoked            Pernambuco   
1672           22-May-2004  2004.0  Unprovoked            Pernambuco   
1675           01-May-2004  2004.0  Unprovoked            Pernambuco   
1685           29-Mar-2004  2004.0  Unprovoked            Pernambuco   
1693           29-Feb-2004  2004.0  Unprovoked            Pernambuco   
1765           24-Jun-2003  2003.0  Unprovoked            Pernambuco   
1778           26-Apr-2003  2003.0    Provoked        Rio de Janeiro   
1779           25-Apr-2003  2003.0  Unprovoked        Rio de Janeiro   
1780           23-Apr-2003  2003.0  Unprovoked            Pernambuco   
1781           23-Apr-2003  2003.0  Unprovoked        Rio de Janeiro   
1807           01-Dec-2002  2002.0  Unprovoked            Pernambuco   
1816           14-Oct-2002  2002.0  Unprovoked            Pernambuco   
1828           16-Sep-2002  2002.0  Unprovoked            Pernambuco   
1844           10-Jul-2002  2002.0  Unprovoked            Pernambuco   
1863           10-May-2002  2002.0  Unprovoked            Pernambuco   
1875           24-Mar-2002  2002.0  Unprovoked            Pernambuco   
1887           03-Jan-2002  2002.0  Unprovoked            Pernambuco   
1967            02-Ap-2001  2001.0  Unprovoked   Rio Grande de Norte   
1972           03-Mar-2001  2001.0  Unprovoked            Pernambuco   
2082           26-Dec-1999  1999.0  Unprovoked            Pernambuco   
2090           06-Nov-1999  1999.0  Unprovoked        Rio de Janeiro   
2124           01-May-1999  1999.0  Unprovoked            Pernambuco   
2129           18-Mar-1999  1999.0    Provoked   Rio Grande de Norte   
2155           02-Nov-1998  1998.0  Unprovoked            Pernambuco   
2159           04-Oct-1998  1998.0  Unprovoked            Pernambuco   
2176           26-Jul-1998  1998.0  Unprovoked            Pernambuco   
2195           01-Apr-1998  1998.0  Unprovoked            Pern

## Investigating 'Year'

In [21]:
attacks_br['Year'].unique()

array([2018., 2017., 2016., 2015., 2014., 2013., 2012., 2011., 2010.,
       2009., 2008., 2007., 2006., 2005., 2004., 2003., 2002., 2001.,
       1999., 1998., 1997., 1996., 1995., 1994., 1993., 1992., 1991.,
       1981., 1980., 1947., 1884., 1826.])

In [22]:
attacks_br[attacks_br['Year'] < 1990]

Date    Year        Type            Area  \
3038              Mar-1981  1981.0  Unprovoked  Rio de Janeiro   
3043              Feb-1981  1981.0  Unprovoked  Rio de Janeiro   
3053           11-Nov-1980  1980.0  Unprovoked      Pernambuco   
3070           28-Mar-1980  1980.0  Unprovoked  Rio de Janeiro   
4569           10-Oct-1947  1947.0  Unprovoked      Pernambuco   
4586              Feb-1947  1947.0  Unprovoked  Rio de Janeiro   
5777  Reported 28-Apr-1884  1884.0  Unprovoked         Unknown   
6101              Dec-1826  1826.0  Unprovoked         Paraiba   

                              Location  \
3038                         Cabo Frio   
3043                         Cabo Frio   
3053                           Piedade   
3070  Copacabana Beach, Rio de Janiero   
4569                           Piedade   
4586                        Copacabana   
5777                             Bahia   
6101                           Paraiba   

                                     Activity                            Name  \
3038                                   Diving                         Unknown   
3043                             Spearfishing                         Unknown   
3053                                 Swimming            Ibson Gomes  Vieira    
3070                                 Swimming                            J.M.   
4569                                Swimming       Father Serafin de Oliveira   
4586                                  Bathing                         Unknown   
5777  Fell overboard from the steamship Chala                            male   
6101                                  Unknown  a seaman from the ship Beverly   

         Sex       Age Fatal (Y/N)     Species  State  
3038  Unknown  Unknown     UNKNOWN  White shark    RJ  
3043  Unknown  Unknown           N      Unknown    RJ  
3053        M       16           Y      Unknown    PE  
3070        M  Unknown           N    2 m shark    RJ  
4569        M       25           Y      Unknown    PE  
4586  Unknown  Unknown           Y      Unknown    RJ  
5777        M  Unknown           Y      Unknown    BA  
6101        M  Unknown           N      Unknown    PB

## Data prior to 1980 seems too unreliable, I will drop it

In [23]:
attacks_br = attacks_br[attacks_br['Year'] >= 1980]

In [24]:
attacks_br

Date    Year        Type                  Area  \
6         03-Jun-2018  2018.0  Unprovoked            Pernambuco   
27        15-Apr-2018  2018.0  Unprovoked            Pernambuco   
51        12-Jan-2018  2018.0  Unprovoked   Fernando de Noronha   
177       06-Feb-2017  2017.0    Provoked            Pernambuco   
181       24-Jan-2017  2017.0  Unprovoked            Pernambuco   
302       02-Mar-2016  2016.0  Unprovoked  Santa Catarina State   
310       02-Feb-2016  2016.0  Unprovoked    Balneário Camboriú   
324       21-Dec-2015  2015.0  Unprovoked            Pernambuco   
543       09-Jun-2014  2014.0  Unprovoked               Unknown   
651       22-Jul-2013  2013.0  Unprovoked            Pernambuco   
673       23-May-2013  2013.0  Unprovoked            Pernambuco   
751       26-Aug-2012  2012.0  Unprovoked            Pernambuco   
845       12-Nov-2011  2011.0  Unprovoked            Pernambuco   
903       29-Jun-2011  2011.0  Unprovoked            Pernambuco   
1047      31-Jan-2010  2010.0  Unprovoked     Rio Grande Do Sul   
1090      13-Sep-2009  2009.0    Provoked            Pernambuco   
1092      07-Sep-2009  2009.0  Unprovoked            Pernambuco   
1246      24-Jun-2008  2008.0  Unprovoked                 Bahia   
1249      11-Jun-2008  2008.0  Unprovoked            Pernambuco   
1253      01-Jun-2008  2008.0  Unprovoked            Pernambuco   
1307      07-Dec-2007  2007.0  Unprovoked            Pernambuco   
1460      10-Jul-2006  2006.0  Unprovoked            Pernambuco   
1471      18-Jun-2006  2006.0  Unprovoked            Pernambuco   
1480      21-May-2006  2006.0  Unprovoked            Pernambuco   
1491      09-Apr-2006  2006.0  Unprovoked            Pernambuco   
1613      04-Feb-2005  2005.0  Unprovoked                 Bahia   
1640      08-Sep-2004  2004.0  Unprovoked            Pernambuco   
1643      21-Aug-2004  2004.0  Unprovoked            Pernambuco   
1667      23-May-2004  2004.0  Unprovoked            Pernambuco   
1672      22-May-2004  2004.0  Unprovoked            Pernambuco   
1675      01-May-2004  2004.0  Unprovoked            Pernambuco   
1685      29-Mar-2004  2004.0  Unprovoked            Pernambuco   
1693      29-Feb-2004  2004.0  Unprovoked            Pernambuco   
1765      24-Jun-2003  2003.0  Unprovoked            Pernambuco   
1778      26-Apr-2003  2003.0    Provoked        Rio de Janeiro   
1779      25-Apr-2003  2003.0  Unprovoked        Rio de Janeiro   
1780      23-Apr-2003  2003.0  Unprovoked            Pernambuco   
1781      23-Apr-2003  2003.0  Unprovoked        Rio de Janeiro   
1807      01-Dec-2002  2002.0  Unprovoked            Pernambuco   
1816      14-Oct-2002  2002.0  Unprovoked            Pernambuco   
1828      16-Sep-2002  2002.0  Unprovoked            Pernambuco   
1844      10-Jul-2002  2002.0  Unprovoked            Pernambuco   
1863      10-May-2002  2002.0  Unprovoked            Pernambuco   
1875      24-Mar-2002  2002.0  Unprovoked            Pernambuco   
1887      03-Jan-2002  2002.0  Unprovoked            Pernambuco   
1967       02-Ap-2001  2001.0  Unprovoked   Rio Grande de Norte   
1972      03-Mar-2001  2001.0  Unprovoked            Pernambuco   
2082      26-Dec-1999  1999.0  Unprovoked            Pernambuco   
2090      06-Nov-1999  1999.0  Unprovoked        Rio de Janeiro   
2124      01-May-1999  1999.0  Unprovoked            Pernambuco   
2129      18-Mar-1999  1999.0    Provoked   Rio Grande de Norte   
2155      02-Nov-1998  1998.0  Unprovoked            Pernambuco   
2159      04-Oct-1998  1998.0  Unprovoked            Pernambuco   
2176      26-Jul-1998  1998.0  Unprovoked            Pernambuco   
2195      01-Apr-1998  1998.0  Unprovoked            Pernambuco   
2223      16-Sep-1997  1997.0  Unprovoked            Pernambuco   
2243      17-Jul-1997  1997.0  Unprovoked            Pernambuco   
2246       2-Jul-1997  1997.0  Unprovoked            Pernambuco   
2250      07-Jun-1997  1997.0  Unprovoked        Rio de Janeiro 

In [25]:
attacks_br['Year'].unique()

array([2018., 2017., 2016., 2015., 2014., 2013., 2012., 2011., 2010.,
       2009., 2008., 2007., 2006., 2005., 2004., 2003., 2002., 2001.,
       1999., 1998., 1997., 1996., 1995., 1994., 1993., 1992., 1991.,
       1981., 1980.])

## I have no need for the 'Area' column any longer so I will drop it

In [26]:
attacks_br = attacks_br.drop('Area',axis=1)

In [27]:
attacks_br

Date    Year        Type  \
6         03-Jun-2018  2018.0  Unprovoked   
27        15-Apr-2018  2018.0  Unprovoked   
51        12-Jan-2018  2018.0  Unprovoked   
177       06-Feb-2017  2017.0    Provoked   
181       24-Jan-2017  2017.0  Unprovoked   
302       02-Mar-2016  2016.0  Unprovoked   
310       02-Feb-2016  2016.0  Unprovoked   
324       21-Dec-2015  2015.0  Unprovoked   
543       09-Jun-2014  2014.0  Unprovoked   
651       22-Jul-2013  2013.0  Unprovoked   
673       23-May-2013  2013.0  Unprovoked   
751       26-Aug-2012  2012.0  Unprovoked   
845       12-Nov-2011  2011.0  Unprovoked   
903       29-Jun-2011  2011.0  Unprovoked   
1047      31-Jan-2010  2010.0  Unprovoked   
1090      13-Sep-2009  2009.0    Provoked   
1092      07-Sep-2009  2009.0  Unprovoked   
1246      24-Jun-2008  2008.0  Unprovoked   
1249      11-Jun-2008  2008.0  Unprovoked   
1253      01-Jun-2008  2008.0  Unprovoked   
1307      07-Dec-2007  2007.0  Unprovoked   
1460      10-Jul-2006  2006.0  Unprovoked   
1471      18-Jun-2006  2006.0  Unprovoked   
1480      21-May-2006  2006.0  Unprovoked   
1491      09-Apr-2006  2006.0  Unprovoked   
1613      04-Feb-2005  2005.0  Unprovoked   
1640      08-Sep-2004  2004.0  Unprovoked   
1643      21-Aug-2004  2004.0  Unprovoked   
1667      23-May-2004  2004.0  Unprovoked   
1672      22-May-2004  2004.0  Unprovoked   
1675      01-May-2004  2004.0  Unprovoked   
1685      29-Mar-2004  2004.0  Unprovoked   
1693      29-Feb-2004  2004.0  Unprovoked   
1765      24-Jun-2003  2003.0  Unprovoked   
1778      26-Apr-2003  2003.0    Provoked   
1779      25-Apr-2003  2003.0  Unprovoked   
1780      23-Apr-2003  2003.0  Unprovoked   
1781      23-Apr-2003  2003.0  Unprovoked   
1807      01-Dec-2002  2002.0  Unprovoked   
1816      14-Oct-2002  2002.0  Unprovoked   
1828      16-Sep-2002  2002.0  Unprovoked   
1844      10-Jul-2002  2002.0  Unprovoked   
1863      10-May-2002  2002.0  Unprovoked   
1875      24-Mar-2002  2002.0  Unprovoked   
1887      03-Jan-2002  2002.0  Unprovoked   
1967       02-Ap-2001  2001.0  Unprovoked   
1972      03-Mar-2001  2001.0  Unprovoked   
2082      26-Dec-1999  1999.0  Unprovoked   
2090      06-Nov-1999  1999.0  Unprovoked   
2124      01-May-1999  1999.0  Unprovoked   
2129      18-Mar-1999  1999.0    Provoked   
2155      02-Nov-1998  1998.0  Unprovoked   
2159      04-Oct-1998  1998.0  Unprovoked   
2176      26-Jul-1998  1998.0  Unprovoked   
2195      01-Apr-1998  1998.0  Unprovoked   
2223      16-Sep-1997  1997.0  Unprovoked   
2243      17-Jul-1997  1997.0  Unprovoked   
2246       2-Jul-1997  1997.0  Unprovoked   
2250      07-Jun-1997  1997.0  Unprovoked   
2257      20-Apr-1997  1997.0  Unprovoked   
2273      28-Oct-1996  1996.0  Unprovoked   
2274      28-Oct-1996  1996.0  Unprovoked   
2279         Oct-1996  1996.0  Unprovoked   
2310      07-Apr-1996  1996.0  Unprovoked   
2349      27-Aug-1995  1995.0  Unprovoked   
2371      07-Jul-1995  1995.0  Unprovoked   
2396      05-Mar-1995  1995.0  Unprovoked   
2400      02-Jan-1995  1995.0  Unprovoked   
2407      13-Dec-1994  1994.0  Unprovoked   
2408      11-Dec-1994  1994.0  Unprovoked   
2412      01-Dec-1994  1994.0  Unprovoked   
2416      18-Oct-1994  1994.0  Unprovoked   
2417      17-Oct-1994  1994.0  Unprovoked   
2425      24-Jul-1994  1994.0  Unprovoked   
2426      09-Jul-1994  1994.0  Unprovoked   
2430      08-Jul-1994  1994.0  Unprovoked   
2447      01-Mar-1994  1994.0  Unprovoked   
2453      31-Jan-1994  1994.0  Unprovoked   
2488      30-Jun-1993  1993.0  Unprovoked   
2500      27-Mar-1993  1993.0  Unprovoked   
2509      23-Jan-1993  1993.0  Unprovoked   
2530      11-Oct-1992  1992.0  Unprovoked   
2532      18-Sep-1992  1992.0  Unprovoked   
2535      10-Sep-1992  1992.0  Unprovoked   
2545      08-Jul-1992  1992.0  Unprovoked   
2546      08-Jul-1992  1992.0  Unprovoked   
2547      28-Jun-1992  1992.0  Unprovoked   
2601      24-Apr-1991  1991.0    Provoked   
3038        

## Checking the columns 'Sex'

In [28]:
attacks_br[attacks_br.columns[6]].unique()

array(['M', 'F', 'Unknown'], dtype=object)

## Actually, 'Name' has information about 'Sex'. Fix that.

In [29]:
attacks_br['Name'].unique()

array(['Jose Ernesto da Silva ', 'Pablo de Melo',
       'Ricardo Ferrari Bulhoes ', 'female', 'Arthur Andrade',
       'Rafael Hermes Thomas', 'Márcio de Castro Palma ', 'Sunarko',
       'Bruna Silva Gobbi ', 'José Rogério da Silva',
       'Tiago José de Oliveira da Silva ', 'Jerônimo Pereira da Paz ',
       'Malisson Lima ', 'Andrei Johann', 'Maurício da Silva Monteiro',
       'Geovanni Tiago Barbosa ', 'Ricardo Garcia Santo Sé\xa0 ',
       'Juan Rodrigues Galvao ', 'Wellington dos Santos ',
       'Malvis Cristino de Souza', 'Unidentified',
       'Humberto Pessoa Batista', 'Rogério Antônio de Carvalho',
       'José Ivair Pereira', 'Antonio de Carvalho Miguel Pereira ',
       'Wagner da Silva', 'Walmir Pereira da Silva',
       'Naiane Barbosa Bringel', 'Orlando Oscar da Silva',
       'Alcindo de Souza Leão Júnior', 'Edimilson Henrique dos Santos',
       'Moses Nunes de Albuquerque Junior', 'male',
       'Felipe Tavares Marinho', 'Tiago Augusto da Silva Machado',
       'A

In [30]:
attacks_br.loc[(attacks_br['Name'] == 'male') | (attacks_br['Name'] == 'female') | (attacks_br['Name'] == 'Unknown')]

Date    Year        Type                          Location  \
177   06-Feb-2017  2017.0    Provoked   Fernando de Noronha Marine Park   
1778  26-Apr-2003  2003.0    Provoked  Copacabana Beach, Rio de Janiero   
1781  23-Apr-2003  2003.0  Unprovoked  Copacabana Beach, Rio de Janiero   
2090  06-Nov-1999  1999.0  Unprovoked     Rio de Janeiro, Guanabara Bay   
3038     Mar-1981  1981.0  Unprovoked                         Cabo Frio   
3043     Feb-1981  1981.0  Unprovoked                         Cabo Frio   

                                   Activity     Name     Sex       Age  \
177             Grabbing shark for a selfie   female        F  Unknown   
1778                        Killing  sharks     male        M  Unknown   
1781                               Swimming     male        M        7   
2090  Spearfishing, but swimming at surface     male        M       39   
3038                                 Diving  Unknown  Unknown  Unknown   
3043                           Spearfishing  Unknown  Unknown  Unknown   

     Fatal (Y/N)            Species  State  
177            N     Lemon shark pup    PE  
1778           N             Unknown    RJ  
1781           N             Unknown    RJ  
2090           N  1.3 to 1.6 m shark    RJ  
3038     UNKNOWN         White shark    RJ  
3043           N             Unknown    RJ

In [31]:
#attacks_br.loc[attacks_br['Name'] == 'Unknown', 'Name'] = 'Unidentified'
attacks_br.loc[((attacks_br['Name'] == 'male') | (attacks_br['Name'] == 'female') | (attacks_br['Name'] == 'Unknown')), 'Name'] = 'Unidentified'

In [32]:
attacks_br['Name'].unique()

array(['Jose Ernesto da Silva ', 'Pablo de Melo',
       'Ricardo Ferrari Bulhoes ', 'Unidentified', 'Arthur Andrade',
       'Rafael Hermes Thomas', 'Márcio de Castro Palma ', 'Sunarko',
       'Bruna Silva Gobbi ', 'José Rogério da Silva',
       'Tiago José de Oliveira da Silva ', 'Jerônimo Pereira da Paz ',
       'Malisson Lima ', 'Andrei Johann', 'Maurício da Silva Monteiro',
       'Geovanni Tiago Barbosa ', 'Ricardo Garcia Santo Sé\xa0 ',
       'Juan Rodrigues Galvao ', 'Wellington dos Santos ',
       'Malvis Cristino de Souza', 'Humberto Pessoa Batista',
       'Rogério Antônio de Carvalho', 'José Ivair Pereira',
       'Antonio de Carvalho Miguel Pereira ', 'Wagner da Silva',
       'Walmir Pereira da Silva', 'Naiane Barbosa Bringel',
       'Orlando Oscar da Silva', 'Alcindo de Souza Leão Júnior',
       'Edimilson Henrique dos Santos',
       'Moses Nunes de Albuquerque Junior', 'Felipe Tavares Marinho',
       'Tiago Augusto da Silva Machado', 'Aylson Gadelha',
       'L

In [33]:
attacks_br.groupby(by=['State']).count()['Date']

State
BA     2
MA     4
PE    72
RJ     9
RN     2
RS     1
SC     2
Name: Date, dtype: int64

In [34]:
attacks_br['Activity'].unique()

array(['Swimming', 'Surfing', 'Grabbing shark for a selfie',
       'Scuba diving', 'Fishing', 'Unknown',
       'Swimming, attempting to rescue a girl  believed to be drowning',
       'Removing fish from a trap', 'Body boarding', 'Bathing', 'Wading',
       'Killing  sharks', 'Swimming ', 'Batin',
       'Spearfishing, but swimming at surface',
       "Scientific research (Dr. Sonny Gruber's student)", 'Windsurfing',
       'Boogie boarding', 'Diving', 'Spearfishing'], dtype=object)

## Cluster 'Activities' into ones that make more sense

In [35]:
attacks_br

Date    Year        Type  \
6         03-Jun-2018  2018.0  Unprovoked   
27        15-Apr-2018  2018.0  Unprovoked   
51        12-Jan-2018  2018.0  Unprovoked   
177       06-Feb-2017  2017.0    Provoked   
181       24-Jan-2017  2017.0  Unprovoked   
302       02-Mar-2016  2016.0  Unprovoked   
310       02-Feb-2016  2016.0  Unprovoked   
324       21-Dec-2015  2015.0  Unprovoked   
543       09-Jun-2014  2014.0  Unprovoked   
651       22-Jul-2013  2013.0  Unprovoked   
673       23-May-2013  2013.0  Unprovoked   
751       26-Aug-2012  2012.0  Unprovoked   
845       12-Nov-2011  2011.0  Unprovoked   
903       29-Jun-2011  2011.0  Unprovoked   
1047      31-Jan-2010  2010.0  Unprovoked   
1090      13-Sep-2009  2009.0    Provoked   
1092      07-Sep-2009  2009.0  Unprovoked   
1246      24-Jun-2008  2008.0  Unprovoked   
1249      11-Jun-2008  2008.0  Unprovoked   
1253      01-Jun-2008  2008.0  Unprovoked   
1307      07-Dec-2007  2007.0  Unprovoked   
1460      10-Jul-2006  2006.0  Unprovoked   
1471      18-Jun-2006  2006.0  Unprovoked   
1480      21-May-2006  2006.0  Unprovoked   
1491      09-Apr-2006  2006.0  Unprovoked   
1613      04-Feb-2005  2005.0  Unprovoked   
1640      08-Sep-2004  2004.0  Unprovoked   
1643      21-Aug-2004  2004.0  Unprovoked   
1667      23-May-2004  2004.0  Unprovoked   
1672      22-May-2004  2004.0  Unprovoked   
1675      01-May-2004  2004.0  Unprovoked   
1685      29-Mar-2004  2004.0  Unprovoked   
1693      29-Feb-2004  2004.0  Unprovoked   
1765      24-Jun-2003  2003.0  Unprovoked   
1778      26-Apr-2003  2003.0    Provoked   
1779      25-Apr-2003  2003.0  Unprovoked   
1780      23-Apr-2003  2003.0  Unprovoked   
1781      23-Apr-2003  2003.0  Unprovoked   
1807      01-Dec-2002  2002.0  Unprovoked   
1816      14-Oct-2002  2002.0  Unprovoked   
1828      16-Sep-2002  2002.0  Unprovoked   
1844      10-Jul-2002  2002.0  Unprovoked   
1863      10-May-2002  2002.0  Unprovoked   
1875      24-Mar-2002  2002.0  Unprovoked   
1887      03-Jan-2002  2002.0  Unprovoked   
1967       02-Ap-2001  2001.0  Unprovoked   
1972      03-Mar-2001  2001.0  Unprovoked   
2082      26-Dec-1999  1999.0  Unprovoked   
2090      06-Nov-1999  1999.0  Unprovoked   
2124      01-May-1999  1999.0  Unprovoked   
2129      18-Mar-1999  1999.0    Provoked   
2155      02-Nov-1998  1998.0  Unprovoked   
2159      04-Oct-1998  1998.0  Unprovoked   
2176      26-Jul-1998  1998.0  Unprovoked   
2195      01-Apr-1998  1998.0  Unprovoked   
2223      16-Sep-1997  1997.0  Unprovoked   
2243      17-Jul-1997  1997.0  Unprovoked   
2246       2-Jul-1997  1997.0  Unprovoked   
2250      07-Jun-1997  1997.0  Unprovoked   
2257      20-Apr-1997  1997.0  Unprovoked   
2273      28-Oct-1996  1996.0  Unprovoked   
2274      28-Oct-1996  1996.0  Unprovoked   
2279         Oct-1996  1996.0  Unprovoked   
2310      07-Apr-1996  1996.0  Unprovoked   
2349      27-Aug-1995  1995.0  Unprovoked   
2371      07-Jul-1995  1995.0  Unprovoked   
2396      05-Mar-1995  1995.0  Unprovoked   
2400      02-Jan-1995  1995.0  Unprovoked   
2407      13-Dec-1994  1994.0  Unprovoked   
2408      11-Dec-1994  1994.0  Unprovoked   
2412      01-Dec-1994  1994.0  Unprovoked   
2416      18-Oct-1994  1994.0  Unprovoked   
2417      17-Oct-1994  1994.0  Unprovoked   
2425      24-Jul-1994  1994.0  Unprovoked   
2426      09-Jul-1994  1994.0  Unprovoked   
2430      08-Jul-1994  1994.0  Unprovoked   
2447      01-Mar-1994  1994.0  Unprovoked   
2453      31-Jan-1994  1994.0  Unprovoked   
2488      30-Jun-1993  1993.0  Unprovoked   
2500      27-Mar-1993  1993.0  Unprovoked   
2509      23-Jan-1993  1993.0  Unprovoked   
2530      11-Oct-1992  1992.0  Unprovoked   
2532      18-Sep-1992  1992.0  Unprovoked   
2535      10-Sep-1992  1992.0  Unprovoked   
2545      08-Jul-1992  1992.0  Unprovoked   
2546      08-Jul-1992  1992.0  Unprovoked   
2547      28-Jun-1992  1992.0  Unprovoked   
2601      24-Apr-1991  1991.0    Provoked   
3038        

In [36]:
attacks_br['Activity'].unique()

array(['Swimming', 'Surfing', 'Grabbing shark for a selfie',
       'Scuba diving', 'Fishing', 'Unknown',
       'Swimming, attempting to rescue a girl  believed to be drowning',
       'Removing fish from a trap', 'Body boarding', 'Bathing', 'Wading',
       'Killing  sharks', 'Swimming ', 'Batin',
       'Spearfishing, but swimming at surface',
       "Scientific research (Dr. Sonny Gruber's student)", 'Windsurfing',
       'Boogie boarding', 'Diving', 'Spearfishing'], dtype=object)

In [37]:
attacks_br.loc[attacks_br['Activity'].str.contains('Swim'), 'Act'] = 'SBW'
attacks_br.loc[attacks_br['Activity'].str.contains('Bat'), 'Act'] = 'SBW'
attacks_br.loc[attacks_br['Activity'].str.contains('Wading'), 'Act'] = 'SBW'
attacks_br.loc[attacks_br['Activity'].str.contains('iving'), 'Act'] = 'Diving'
attacks_br.loc[attacks_br['Activity'].str.contains('urfing'), 'Act'] = 'Surfing'
attacks_br.loc[attacks_br['Activity'].str.contains('boarding'), 'Act'] = 'Surfing'
attacks_br.loc[attacks_br['Activity'].str.contains('ish'), 'Act'] = 'Fishing'
attacks_br.loc[attacks_br['Activity'].str.contains('research'), 'Act'] = 'Scientific Research'
attacks_br.loc[attacks_br['Act'].isna(), 'Act'] = 'Others'

In [38]:
attacks_br

Date    Year        Type  \
6         03-Jun-2018  2018.0  Unprovoked   
27        15-Apr-2018  2018.0  Unprovoked   
51        12-Jan-2018  2018.0  Unprovoked   
177       06-Feb-2017  2017.0    Provoked   
181       24-Jan-2017  2017.0  Unprovoked   
302       02-Mar-2016  2016.0  Unprovoked   
310       02-Feb-2016  2016.0  Unprovoked   
324       21-Dec-2015  2015.0  Unprovoked   
543       09-Jun-2014  2014.0  Unprovoked   
651       22-Jul-2013  2013.0  Unprovoked   
673       23-May-2013  2013.0  Unprovoked   
751       26-Aug-2012  2012.0  Unprovoked   
845       12-Nov-2011  2011.0  Unprovoked   
903       29-Jun-2011  2011.0  Unprovoked   
1047      31-Jan-2010  2010.0  Unprovoked   
1090      13-Sep-2009  2009.0    Provoked   
1092      07-Sep-2009  2009.0  Unprovoked   
1246      24-Jun-2008  2008.0  Unprovoked   
1249      11-Jun-2008  2008.0  Unprovoked   
1253      01-Jun-2008  2008.0  Unprovoked   
1307      07-Dec-2007  2007.0  Unprovoked   
1460      10-Jul-2006  2006.0  Unprovoked   
1471      18-Jun-2006  2006.0  Unprovoked   
1480      21-May-2006  2006.0  Unprovoked   
1491      09-Apr-2006  2006.0  Unprovoked   
1613      04-Feb-2005  2005.0  Unprovoked   
1640      08-Sep-2004  2004.0  Unprovoked   
1643      21-Aug-2004  2004.0  Unprovoked   
1667      23-May-2004  2004.0  Unprovoked   
1672      22-May-2004  2004.0  Unprovoked   
1675      01-May-2004  2004.0  Unprovoked   
1685      29-Mar-2004  2004.0  Unprovoked   
1693      29-Feb-2004  2004.0  Unprovoked   
1765      24-Jun-2003  2003.0  Unprovoked   
1778      26-Apr-2003  2003.0    Provoked   
1779      25-Apr-2003  2003.0  Unprovoked   
1780      23-Apr-2003  2003.0  Unprovoked   
1781      23-Apr-2003  2003.0  Unprovoked   
1807      01-Dec-2002  2002.0  Unprovoked   
1816      14-Oct-2002  2002.0  Unprovoked   
1828      16-Sep-2002  2002.0  Unprovoked   
1844      10-Jul-2002  2002.0  Unprovoked   
1863      10-May-2002  2002.0  Unprovoked   
1875      24-Mar-2002  2002.0  Unprovoked   
1887      03-Jan-2002  2002.0  Unprovoked   
1967       02-Ap-2001  2001.0  Unprovoked   
1972      03-Mar-2001  2001.0  Unprovoked   
2082      26-Dec-1999  1999.0  Unprovoked   
2090      06-Nov-1999  1999.0  Unprovoked   
2124      01-May-1999  1999.0  Unprovoked   
2129      18-Mar-1999  1999.0    Provoked   
2155      02-Nov-1998  1998.0  Unprovoked   
2159      04-Oct-1998  1998.0  Unprovoked   
2176      26-Jul-1998  1998.0  Unprovoked   
2195      01-Apr-1998  1998.0  Unprovoked   
2223      16-Sep-1997  1997.0  Unprovoked   
2243      17-Jul-1997  1997.0  Unprovoked   
2246       2-Jul-1997  1997.0  Unprovoked   
2250      07-Jun-1997  1997.0  Unprovoked   
2257      20-Apr-1997  1997.0  Unprovoked   
2273      28-Oct-1996  1996.0  Unprovoked   
2274      28-Oct-1996  1996.0  Unprovoked   
2279         Oct-1996  1996.0  Unprovoked   
2310      07-Apr-1996  1996.0  Unprovoked   
2349      27-Aug-1995  1995.0  Unprovoked   
2371      07-Jul-1995  1995.0  Unprovoked   
2396      05-Mar-1995  1995.0  Unprovoked   
2400      02-Jan-1995  1995.0  Unprovoked   
2407      13-Dec-1994  1994.0  Unprovoked   
2408      11-Dec-1994  1994.0  Unprovoked   
2412      01-Dec-1994  1994.0  Unprovoked   
2416      18-Oct-1994  1994.0  Unprovoked   
2417      17-Oct-1994  1994.0  Unprovoked   
2425      24-Jul-1994  1994.0  Unprovoked   
2426      09-Jul-1994  1994.0  Unprovoked   
2430      08-Jul-1994  1994.0  Unprovoked   
2447      01-Mar-1994  1994.0  Unprovoked   
2453      31-Jan-1994  1994.0  Unprovoked   
2488      30-Jun-1993  1993.0  Unprovoked   
2500      27-Mar-1993  1993.0  Unprovoked   
2509      23-Jan-1993  1993.0  Unprovoked   
2530      11-Oct-1992  1992.0  Unprovoked   
2532      18-Sep-1992  1992.0  Unprovoked   
2535      10-Sep-1992  1992.0  Unprovoked   
2545      08-Jul-1992  1992.0  Unprovoked   
2546      08-Jul-1992  1992.0  Unprovoked   
2547      28-Jun-1992  1992.0  Unprovoked   
2601      24-Apr-1991  1991.0    Provoked   
3038        

In [39]:
attacks_br['Act'].unique()

array(['SBW', 'Surfing', 'Others', 'Diving', 'Fishing',
       'Scientific Research'], dtype=object)

Grouping activities while attacked in six major ones. 
SBW stands for 'Swimming, Bathing and Wading'. Now I can drop the old 'Activity' column.

In [40]:
attacks_br = attacks_br.drop(attacks_br.columns[4], axis=1)
attacks_br.head(3)

Date    Year        Type               Location  \
6   03-Jun-2018  2018.0  Unprovoked  Piedade Beach, Recife   
27  15-Apr-2018  2018.0  Unprovoked                Piedade   
51  12-Jan-2018  2018.0  Unprovoked        Conceicao Beach   

                        Name Sex  Age Fatal (Y/N)     Species  State      Act  
6     Jose Ernesto da Silva     M  18           Y  Tiger shark    PE      SBW  
27             Pablo de Melo    M  34     Unknown      Unknown    PE      SBW  
51  Ricardo Ferrari Bulhoes     M  20           N  small shark    PE  Surfing

In [41]:
attacks_br.rename(columns={'Act':'Activity'}, inplace=True) #renaming new 'Activity' column
attacks_br.head(3)

Date    Year        Type               Location  \
6   03-Jun-2018  2018.0  Unprovoked  Piedade Beach, Recife   
27  15-Apr-2018  2018.0  Unprovoked                Piedade   
51  12-Jan-2018  2018.0  Unprovoked        Conceicao Beach   

                        Name Sex  Age Fatal (Y/N)     Species  State Activity  
6     Jose Ernesto da Silva     M  18           Y  Tiger shark    PE      SBW  
27             Pablo de Melo    M  34     Unknown      Unknown    PE      SBW  
51  Ricardo Ferrari Bulhoes     M  20           N  small shark    PE  Surfing

In [42]:
attacks_br = attacks_br.drop(attacks_br.columns[8], axis=1) #dropping 'Species' column as I will not need it
attacks_br.head(3)

Date    Year        Type               Location  \
6   03-Jun-2018  2018.0  Unprovoked  Piedade Beach, Recife   
27  15-Apr-2018  2018.0  Unprovoked                Piedade   
51  12-Jan-2018  2018.0  Unprovoked        Conceicao Beach   

                        Name Sex  Age Fatal (Y/N) State Activity  
6     Jose Ernesto da Silva     M  18           Y    PE      SBW  
27             Pablo de Melo    M  34     Unknown    PE      SBW  
51  Ricardo Ferrari Bulhoes     M  20           N    PE  Surfing

In [43]:
attacks_br.columns

Index(['Date', 'Year', 'Type', 'Location', 'Name', 'Sex ', 'Age',
       'Fatal (Y/N)', 'State', 'Activity'],
      dtype='object')

In [44]:
#Fixing columns names
attacks_br.rename(columns={'Sex ':'Sex','Fatal (Y/N)':'Fatal'}, inplace=True)
attacks_br.head(3)

Date    Year        Type               Location  \
6   03-Jun-2018  2018.0  Unprovoked  Piedade Beach, Recife   
27  15-Apr-2018  2018.0  Unprovoked                Piedade   
51  12-Jan-2018  2018.0  Unprovoked        Conceicao Beach   

                        Name Sex Age    Fatal State Activity  
6     Jose Ernesto da Silva    M  18        Y    PE      SBW  
27             Pablo de Melo   M  34  Unknown    PE      SBW  
51  Ricardo Ferrari Bulhoes    M  20        N    PE  Surfing

Now, I will check for every row in each column has information that makes sense

## Date

In [45]:
attacks_br['Date'].unique()

array(['03-Jun-2018', '15-Apr-2018', '12-Jan-2018', '06-Feb-2017',
       '24-Jan-2017', '02-Mar-2016', '02-Feb-2016', '21-Dec-2015',
       '09-Jun-2014', '    22-Jul-2013', '23-May-2013', '26-Aug-2012',
       '12-Nov-2011', '29-Jun-2011', '31-Jan-2010', '13-Sep-2009',
       '07-Sep-2009', '24-Jun-2008', '11-Jun-2008', '01-Jun-2008',
       '07-Dec-2007', '10-Jul-2006', '18-Jun-2006', '21-May-2006',
       '09-Apr-2006', '04-Feb-2005', '08-Sep-2004', '21-Aug-2004',
       '23-May-2004', '22-May-2004', '01-May-2004', '29-Mar-2004',
       '29-Feb-2004', '24-Jun-2003', '26-Apr-2003', '25-Apr-2003',
       '23-Apr-2003', '01-Dec-2002', '14-Oct-2002', '16-Sep-2002',
       '10-Jul-2002', '10-May-2002', '24-Mar-2002', '03-Jan-2002',
       '02-Ap-2001', '03-Mar-2001', '26-Dec-1999', '06-Nov-1999',
       '01-May-1999', '18-Mar-1999', '02-Nov-1998', '04-Oct-1998',
       '26-Jul-1998', '01-Apr-1998', '16-Sep-1997', '17-Jul-1997',
       ' 2-Jul-1997', '07-Jun-1997', '20-Apr-1997', '28-Oct

In [46]:
#'    22-Jul-2013' seems weird but now I want to have month columns

In [47]:
#df["Order_Date"] = df["Shipment ID"].str.extract(r"(\d{8})")

In [48]:
import re

In [49]:
#re.findall("[a-zA-Z]+", st)

In [50]:
#re.findall('[a-zA-Z]+', attacks_br['Date'])

In [51]:
#re.findall('[^A-Za-z]',attacks_br['Date'])

In [52]:
def find_alpha(text):
    result = re.findall('[a-zA-Z]+',text)
    result = ''.join(result)
    return result

In [53]:
# df['nonalpha']=df['company_code'].apply(lambda x: find_nonalpha(x))
attacks_br['Month'] = attacks_br['Date'].apply(lambda x: find_alpha(x))

In [54]:
#attacks_br.head(10)

In [55]:
attacks_br['Month'].unique()

array(['Jun', 'Apr', 'Jan', 'Feb', 'Mar', 'Dec', 'Jul', 'May', 'Aug',
       'Nov', 'Sep', 'Oct', 'Ap'], dtype=object)

In [56]:
attacks_br.head(10)

Date    Year        Type                            Location  \
6        03-Jun-2018  2018.0  Unprovoked               Piedade Beach, Recife   
27       15-Apr-2018  2018.0  Unprovoked                             Piedade   
51       12-Jan-2018  2018.0  Unprovoked                     Conceicao Beach   
177      06-Feb-2017  2017.0    Provoked     Fernando de Noronha Marine Park   
181      24-Jan-2017  2017.0  Unprovoked                  Boa Viagem, Recife   
302      02-Mar-2016  2016.0  Unprovoked  Escalerio Beach Balneário Camboriú   
310      02-Feb-2016  2016.0  Unprovoked                  Estaleirinho beach   
324      21-Dec-2015  2015.0  Unprovoked                  Fernano de Noronha   
543      09-Jun-2014  2014.0  Unprovoked  500 km off the coast of Pernambuco   
651      22-Jul-2013  2013.0  Unprovoked            Boa Viagem Beach, Recife   

                         Name Sex      Age    Fatal State Activity Month  
6      Jose Ernesto da Silva    M       18        Y    PE      SBW   Jun  
27              Pablo de Melo   M       34  Unknown    PE      SBW   Apr  
51   Ricardo Ferrari Bulhoes    M       20        N    PE  Surfing   Jan  
177              Unidentified   F  Unknown        N    PE   Others   Feb  
181            Arthur Andrade   M       21        N    PE      SBW   Jan  
302      Rafael Hermes Thomas   M       41        N    SC      SBW   Mar  
310      Rafael Hermes Thomas   M       41        N    SC      SBW   Feb  
324   Márcio de Castro Palma    M       32        N    PE   Diving   Dec  
543                   Sunarko   M       43        N    PE  Fishing   Jun  
651        Bruna Silva Gobbi    F       18        Y    PE      SBW   Jul

In [57]:
#attacks_br.set_axis(['Date','Month','Year','Type','Location','Name','Sex','Age','Fatal','State','Activity'], axis=1, inplace=False)

In [58]:
attacks_br = attacks_br[['Date','Month','Year','Type','Location','Name','Sex','Age','Fatal','State','Activity']]

In [59]:
attacks_br.head()

Date Month    Year        Type                         Location  \
6    03-Jun-2018   Jun  2018.0  Unprovoked            Piedade Beach, Recife   
27   15-Apr-2018   Apr  2018.0  Unprovoked                          Piedade   
51   12-Jan-2018   Jan  2018.0  Unprovoked                  Conceicao Beach   
177  06-Feb-2017   Feb  2017.0    Provoked  Fernando de Noronha Marine Park   
181  24-Jan-2017   Jan  2017.0  Unprovoked               Boa Viagem, Recife   

                         Name Sex      Age    Fatal State Activity  
6      Jose Ernesto da Silva    M       18        Y    PE      SBW  
27              Pablo de Melo   M       34  Unknown    PE      SBW  
51   Ricardo Ferrari Bulhoes    M       20        N    PE  Surfing  
177              Unidentified   F  Unknown        N    PE   Others  
181            Arthur Andrade   M       21        N    PE      SBW

In [60]:
attacks_br['Date'].unique()

array(['03-Jun-2018', '15-Apr-2018', '12-Jan-2018', '06-Feb-2017',
       '24-Jan-2017', '02-Mar-2016', '02-Feb-2016', '21-Dec-2015',
       '09-Jun-2014', '    22-Jul-2013', '23-May-2013', '26-Aug-2012',
       '12-Nov-2011', '29-Jun-2011', '31-Jan-2010', '13-Sep-2009',
       '07-Sep-2009', '24-Jun-2008', '11-Jun-2008', '01-Jun-2008',
       '07-Dec-2007', '10-Jul-2006', '18-Jun-2006', '21-May-2006',
       '09-Apr-2006', '04-Feb-2005', '08-Sep-2004', '21-Aug-2004',
       '23-May-2004', '22-May-2004', '01-May-2004', '29-Mar-2004',
       '29-Feb-2004', '24-Jun-2003', '26-Apr-2003', '25-Apr-2003',
       '23-Apr-2003', '01-Dec-2002', '14-Oct-2002', '16-Sep-2002',
       '10-Jul-2002', '10-May-2002', '24-Mar-2002', '03-Jan-2002',
       '02-Ap-2001', '03-Mar-2001', '26-Dec-1999', '06-Nov-1999',
       '01-May-1999', '18-Mar-1999', '02-Nov-1998', '04-Oct-1998',
       '26-Jul-1998', '01-Apr-1998', '16-Sep-1997', '17-Jul-1997',
       ' 2-Jul-1997', '07-Jun-1997', '20-Apr-1997', '28-Oct

In [61]:
attacks_br.loc[attacks_br['Date'] != '', 'Date'] = attacks_br['Date'].str.strip()

In [62]:
attacks_br['Date'].unique()

array(['03-Jun-2018', '15-Apr-2018', '12-Jan-2018', '06-Feb-2017',
       '24-Jan-2017', '02-Mar-2016', '02-Feb-2016', '21-Dec-2015',
       '09-Jun-2014', '22-Jul-2013', '23-May-2013', '26-Aug-2012',
       '12-Nov-2011', '29-Jun-2011', '31-Jan-2010', '13-Sep-2009',
       '07-Sep-2009', '24-Jun-2008', '11-Jun-2008', '01-Jun-2008',
       '07-Dec-2007', '10-Jul-2006', '18-Jun-2006', '21-May-2006',
       '09-Apr-2006', '04-Feb-2005', '08-Sep-2004', '21-Aug-2004',
       '23-May-2004', '22-May-2004', '01-May-2004', '29-Mar-2004',
       '29-Feb-2004', '24-Jun-2003', '26-Apr-2003', '25-Apr-2003',
       '23-Apr-2003', '01-Dec-2002', '14-Oct-2002', '16-Sep-2002',
       '10-Jul-2002', '10-May-2002', '24-Mar-2002', '03-Jan-2002',
       '02-Ap-2001', '03-Mar-2001', '26-Dec-1999', '06-Nov-1999',
       '01-May-1999', '18-Mar-1999', '02-Nov-1998', '04-Oct-1998',
       '26-Jul-1998', '01-Apr-1998', '16-Sep-1997', '17-Jul-1997',
       '2-Jul-1997', '07-Jun-1997', '20-Apr-1997', '28-Oct-1996

In [63]:
attacks_br.head(10) #'Date' is ok

Date Month    Year        Type  \
6    03-Jun-2018   Jun  2018.0  Unprovoked   
27   15-Apr-2018   Apr  2018.0  Unprovoked   
51   12-Jan-2018   Jan  2018.0  Unprovoked   
177  06-Feb-2017   Feb  2017.0    Provoked   
181  24-Jan-2017   Jan  2017.0  Unprovoked   
302  02-Mar-2016   Mar  2016.0  Unprovoked   
310  02-Feb-2016   Feb  2016.0  Unprovoked   
324  21-Dec-2015   Dec  2015.0  Unprovoked   
543  09-Jun-2014   Jun  2014.0  Unprovoked   
651  22-Jul-2013   Jul  2013.0  Unprovoked   

                               Location                      Name Sex  \
6                 Piedade Beach, Recife    Jose Ernesto da Silva    M   
27                              Piedade             Pablo de Melo   M   
51                      Conceicao Beach  Ricardo Ferrari Bulhoes    M   
177     Fernando de Noronha Marine Park              Unidentified   F   
181                  Boa Viagem, Recife            Arthur Andrade   M   
302  Escalerio Beach Balneário Camboriú      Rafael Hermes Thomas   M   
310                  Estaleirinho beach      Rafael Hermes Thomas   M   
324                  Fernano de Noronha   Márcio de Castro Palma    M   
543  500 km off the coast of Pernambuco                   Sunarko   M   
651            Boa Viagem Beach, Recife        Bruna Silva Gobbi    F   

         Age    Fatal State Activity  
6         18        Y    PE      SBW  
27        34  Unknown    PE      SBW  
51        20        N    PE  Surfing  
177  Unknown        N    PE   Others  
181       21        N    PE      SBW  
302       41        N    SC      SBW  
310       41        N    SC      SBW  
324       32        N    PE   Diving  
543       43        N    PE  Fishing  
651       18        Y    PE      SBW

## Year

In [64]:
attacks_br['Year'].unique()

array([2018., 2017., 2016., 2015., 2014., 2013., 2012., 2011., 2010.,
       2009., 2008., 2007., 2006., 2005., 2004., 2003., 2002., 2001.,
       1999., 1998., 1997., 1996., 1995., 1994., 1993., 1992., 1991.,
       1981., 1980.])

## Type

In [65]:
attacks_br['Type'].unique()

array(['Unprovoked', 'Provoked'], dtype=object)

## Location

In [66]:
attacks_br['Location'].unique()

array(['Piedade Beach, Recife', 'Piedade', 'Conceicao Beach',
       'Fernando de Noronha Marine Park', 'Boa Viagem, Recife',
       'Escalerio Beach Balneário Camboriú', 'Estaleirinho beach',
       'Fernano de Noronha', '500 km off the coast of Pernambuco',
       'Boa Viagem Beach, Recife', 'Coral Cove Beach',
       'Coral Cove, Cabo de Santo Agostinho',
       'Punta Del Chifre Beach, Olinda', 'Praia do Pina',
       'Atlantis Beach, near Tramandai', 'Piedade, Recife', 'Guarajuba',
       'Punta Del Chifre, Olinda', 'Itamaracá',
       'Body recovered at Goiana', 'Ilhéus', 'Pina, Recife',
       'Piedade Beach', 'Piedade ', 'Copacabana Beach, Rio de Janiero',
       'Pau Amarelo Beach, Paulista District (17 km from Recife)',
       'Piedade Beach, Jaboatão dos Guararapes City, Recife',
       'Piedade Beach, Jaboatão dos Guararapes City', 'Candeias',
       'Praia de Camapum, Macau', 'Rio de Janeiro, Guanabara Bay',
       'Atol das Rochas', ' Boa Viagem Beach, Recife', 'Paiva',
 

I will create a dict to relate location to a city in Brazil 

In [67]:
myset = set(attacks_br['Location'])
keys_list = list(myset)
keys_list.sort()
values_list = ['Recife', 'High Sea', 'Tramandaí', 'Atol das Rocas', 'Jaboatão dos Guararapes', 'Recife', 'Recife', 'Goiana', 'Cabo Frio', 'Jaboatão dos Guararapes', 'Fernando de Noronha', 'Rio de Janeiro', 'Rio de Janeiro', 'Cabo de Santo Agostinho',
              'Cabo de Santo Agostinho', 'Balneário Camboriú', 'Balneário Camboriú', 'Fernando de Noronha', 'Fernando de Noronha', 'Guarajuba', 'Ilhéus', 'Itamaracá', 'Armação dos Búzios', 'Cabo de Santo Agostinho', 'Paulista', 'Jaboatão dos Guararapes',
              'Jaboatão dos Guararapes','Jaboatão dos Guararapes','Jaboatão dos Guararapes','Jaboatão dos Guararapes','Jaboatão dos Guararapes','Jaboatão dos Guararapes', 'Recife', 'São Luís', 'Macau', 'Paulista', 'São Luís', 'São Luís', 'Recife', 
              'Olinda', 'Olinda', 'Rio de Janeiro', 'Unknown']
zip_iterator = zip(keys_list, values_list)

a_dictionary = dict(zip_iterator)

print(a_dictionary)

{' Boa Viagem Beach, Recife': 'Recife', '500 km off the coast of Pernambuco': 'High Sea', 'Atlantis Beach, near Tramandai': 'Tramandaí', 'Atol das Rochas': 'Atol das Rocas', 'Barra de Jangada': 'Jaboatão dos Guararapes', 'Boa Viagem Beach, Recife': 'Recife', 'Boa Viagem, Recife': 'Recife', 'Body recovered at Goiana': 'Goiana', 'Cabo Frio': 'Cabo Frio', 'Candeias': 'Jaboatão dos Guararapes', 'Conceicao Beach': 'Fernando de Noronha', 'Copacabana Beach, Rio de Janiero': 'Rio de Janeiro', 'Copacabana, Rio de Janeiro': 'Rio de Janeiro', 'Coral Cove Beach': 'Cabo de Santo Agostinho', 'Coral Cove, Cabo de Santo Agostinho': 'Cabo de Santo Agostinho', 'Escalerio Beach Balneário Camboriú': 'Balneário Camboriú', 'Estaleirinho beach': 'Balneário Camboriú', 'Fernando de Noronha Marine Park': 'Fernando de Noronha', 'Fernano de Noronha': 'Fernando de Noronha', 'Guarajuba': 'Guarajuba', 'Ilhéus': 'Ilhéus', 'Itamaracá': 'Itamaracá', 'Manguinhos, Ilha Feia, Búzios': 'Armação dos Búzios', 'Paiva': 'Cabo 

In [68]:
attacks_br['City'] = attacks_br['Location'].map(a_dictionary)

In [69]:
attacks_br.head()

Date Month    Year        Type                         Location  \
6    03-Jun-2018   Jun  2018.0  Unprovoked            Piedade Beach, Recife   
27   15-Apr-2018   Apr  2018.0  Unprovoked                          Piedade   
51   12-Jan-2018   Jan  2018.0  Unprovoked                  Conceicao Beach   
177  06-Feb-2017   Feb  2017.0    Provoked  Fernando de Noronha Marine Park   
181  24-Jan-2017   Jan  2017.0  Unprovoked               Boa Viagem, Recife   

                         Name Sex      Age    Fatal State Activity  \
6      Jose Ernesto da Silva    M       18        Y    PE      SBW   
27              Pablo de Melo   M       34  Unknown    PE      SBW   
51   Ricardo Ferrari Bulhoes    M       20        N    PE  Surfing   
177              Unidentified   F  Unknown        N    PE   Others   
181            Arthur Andrade   M       21        N    PE      SBW   

                        City  
6    Jaboatão dos Guararapes  
27   Jaboatão dos Guararapes  
51       Fernando de Noronha  
177      Fernando de Noronha  
181                   Recife

In [70]:
attacks_br['City'].unique()

array(['Jaboatão dos Guararapes', 'Fernando de Noronha', 'Recife',
       'Balneário Camboriú', 'High Sea', 'Cabo de Santo Agostinho',
       'Olinda', 'Tramandaí', 'Guarajuba', 'Itamaracá', 'Goiana',
       'Ilhéus', 'Rio de Janeiro', 'Paulista', 'Macau', 'Atol das Rocas',
       'Armação dos Búzios', 'Unknown', 'São Luís', 'Cabo Frio'],
      dtype=object)

In [71]:
# attacks_br.loc[attacks_br['Date'] != '', 'Date'] = attacks_br['Date'].str.strip()

attacks_br.loc[attacks_br['Location'] == 'Unknown']

Date Month    Year        Type Location           Name      Sex  \
2279  Oct-1996   Oct  1996.0  Unprovoked  Unknown  Josebias Dias  Unknown   

          Age Fatal State Activity     City  
2279  Unknown     N    PE   Others  Unknown

In [72]:
attacks_br = attacks_br[['Date','Month','Year','Location','City','State','Name','Sex','Age','Activity','Type','Fatal']]

In [73]:
attacks_br.head(10)

Date Month    Year                            Location  \
6    03-Jun-2018   Jun  2018.0               Piedade Beach, Recife   
27   15-Apr-2018   Apr  2018.0                             Piedade   
51   12-Jan-2018   Jan  2018.0                     Conceicao Beach   
177  06-Feb-2017   Feb  2017.0     Fernando de Noronha Marine Park   
181  24-Jan-2017   Jan  2017.0                  Boa Viagem, Recife   
302  02-Mar-2016   Mar  2016.0  Escalerio Beach Balneário Camboriú   
310  02-Feb-2016   Feb  2016.0                  Estaleirinho beach   
324  21-Dec-2015   Dec  2015.0                  Fernano de Noronha   
543  09-Jun-2014   Jun  2014.0  500 km off the coast of Pernambuco   
651  22-Jul-2013   Jul  2013.0            Boa Viagem Beach, Recife   

                        City State                      Name Sex      Age  \
6    Jaboatão dos Guararapes    PE    Jose Ernesto da Silva    M       18   
27   Jaboatão dos Guararapes    PE             Pablo de Melo   M       34   
51       Fernando de Noronha    PE  Ricardo Ferrari Bulhoes    M       20   
177      Fernando de Noronha    PE              Unidentified   F  Unknown   
181                   Recife    PE            Arthur Andrade   M       21   
302       Balneário Camboriú    SC      Rafael Hermes Thomas   M       41   
310       Balneário Camboriú    SC      Rafael Hermes Thomas   M       41   
324      Fernando de Noronha    PE   Márcio de Castro Palma    M       32   
543                 High Sea    PE                   Sunarko   M       43   
651                   Recife    PE        Bruna Silva Gobbi    F       18   

    Activity        Type    Fatal  
6        SBW  Unprovoked        Y  
27       SBW  Unprovoked  Unknown  
51   Surfing  Unprovoked        N  
177   Others    Provoked        N  
181      SBW  Unprovoked        N  
302      SBW  Unprovoked        N  
310      SBW  Unprovoked        N  
324   Diving  Unprovoked        N  
543  Fishing  Unprovoked        N  
651      SBW  Unprovoked        Y

## Name

In [74]:
attacks_br['Name'].unique()

array(['Jose Ernesto da Silva ', 'Pablo de Melo',
       'Ricardo Ferrari Bulhoes ', 'Unidentified', 'Arthur Andrade',
       'Rafael Hermes Thomas', 'Márcio de Castro Palma ', 'Sunarko',
       'Bruna Silva Gobbi ', 'José Rogério da Silva',
       'Tiago José de Oliveira da Silva ', 'Jerônimo Pereira da Paz ',
       'Malisson Lima ', 'Andrei Johann', 'Maurício da Silva Monteiro',
       'Geovanni Tiago Barbosa ', 'Ricardo Garcia Santo Sé\xa0 ',
       'Juan Rodrigues Galvao ', 'Wellington dos Santos ',
       'Malvis Cristino de Souza', 'Humberto Pessoa Batista',
       'Rogério Antônio de Carvalho', 'José Ivair Pereira',
       'Antonio de Carvalho Miguel Pereira ', 'Wagner da Silva',
       'Walmir Pereira da Silva', 'Naiane Barbosa Bringel',
       'Orlando Oscar da Silva', 'Alcindo de Souza Leão Júnior',
       'Edimilson Henrique dos Santos',
       'Moses Nunes de Albuquerque Junior', 'Felipe Tavares Marinho',
       'Tiago Augusto da Silva Machado', 'Aylson Gadelha',
       'L

## Sex

In [75]:
attacks_br['Sex'].unique()

array(['M', 'F', 'Unknown'], dtype=object)

## Age

In [76]:
attacks_br['Age'].unique()

array(['18', '34', '20', 'Unknown', '21', '41', '32', '43', '35', '29',
       '15', '14', '28', '27', '33', '12', '24', '17', '22', '16', '7',
       '19', '36', '40', '39', '25'], dtype=object)

## Activity

In [77]:
attacks_br['Type'].unique()

array(['Unprovoked', 'Provoked'], dtype=object)

## Activity

In [78]:
attacks_br['Activity'].unique()

array(['SBW', 'Surfing', 'Others', 'Diving', 'Fishing',
       'Scientific Research'], dtype=object)

In [79]:
attacks_br

Date Month    Year  \
6     03-Jun-2018   Jun  2018.0   
27    15-Apr-2018   Apr  2018.0   
51    12-Jan-2018   Jan  2018.0   
177   06-Feb-2017   Feb  2017.0   
181   24-Jan-2017   Jan  2017.0   
302   02-Mar-2016   Mar  2016.0   
310   02-Feb-2016   Feb  2016.0   
324   21-Dec-2015   Dec  2015.0   
543   09-Jun-2014   Jun  2014.0   
651   22-Jul-2013   Jul  2013.0   
673   23-May-2013   May  2013.0   
751   26-Aug-2012   Aug  2012.0   
845   12-Nov-2011   Nov  2011.0   
903   29-Jun-2011   Jun  2011.0   
1047  31-Jan-2010   Jan  2010.0   
1090  13-Sep-2009   Sep  2009.0   
1092  07-Sep-2009   Sep  2009.0   
1246  24-Jun-2008   Jun  2008.0   
1249  11-Jun-2008   Jun  2008.0   
1253  01-Jun-2008   Jun  2008.0   
1307  07-Dec-2007   Dec  2007.0   
1460  10-Jul-2006   Jul  2006.0   
1471  18-Jun-2006   Jun  2006.0   
1480  21-May-2006   May  2006.0   
1491  09-Apr-2006   Apr  2006.0   
1613  04-Feb-2005   Feb  2005.0   
1640  08-Sep-2004   Sep  2004.0   
1643  21-Aug-2004   Aug  2004.0   
1667  23-May-2004   May  2004.0   
1672  22-May-2004   May  2004.0   
1675  01-May-2004   May  2004.0   
1685  29-Mar-2004   Mar  2004.0   
1693  29-Feb-2004   Feb  2004.0   
1765  24-Jun-2003   Jun  2003.0   
1778  26-Apr-2003   Apr  2003.0   
1779  25-Apr-2003   Apr  2003.0   
1780  23-Apr-2003   Apr  2003.0   
1781  23-Apr-2003   Apr  2003.0   
1807  01-Dec-2002   Dec  2002.0   
1816  14-Oct-2002   Oct  2002.0   
1828  16-Sep-2002   Sep  2002.0   
1844  10-Jul-2002   Jul  2002.0   
1863  10-May-2002   May  2002.0   
1875  24-Mar-2002   Mar  2002.0   
1887  03-Jan-2002   Jan  2002.0   
1967   02-Ap-2001    Ap  2001.0   
1972  03-Mar-2001   Mar  2001.0   
2082  26-Dec-1999   Dec  1999.0   
2090  06-Nov-1999   Nov  1999.0   
2124  01-May-1999   May  1999.0   
2129  18-Mar-1999   Mar  1999.0   
2155  02-Nov-1998   Nov  1998.0   
2159  04-Oct-1998   Oct  1998.0   
2176  26-Jul-1998   Jul  1998.0   
2195  01-Apr-1998   Apr  1998.0   
2223  16-Sep-1997   Sep  1997.0   
2243  17-Jul-1997   Jul  1997.0   
2246   2-Jul-1997   Jul  1997.0   
2250  07-Jun-1997   Jun  1997.0   
2257  20-Apr-1997   Apr  1997.0   
2273  28-Oct-1996   Oct  1996.0   
2274  28-Oct-1996   Oct  1996.0   
2279     Oct-1996   Oct  1996.0   
2310  07-Apr-1996   Apr  1996.0   
2349  27-Aug-1995   Aug  1995.0   
2371  07-Jul-1995   Jul  1995.0   
2396  05-Mar-1995   Mar  1995.0   
2400  02-Jan-1995   Jan  1995.0   
2407  13-Dec-1994   Dec  1994.0   
2408  11-Dec-1994   Dec  1994.0   
2412  01-Dec-1994   Dec  1994.0   
2416  18-Oct-1994   Oct  1994.0   
2417  17-Oct-1994   Oct  1994.0   
2425  24-Jul-1994   Jul  1994.0   
2426  09-Jul-1994   Jul  1994.0   
2430  08-Jul-1994   Jul  1994.0   
2447  01-Mar-1994   Mar  1994.0   
2453  31-Jan-1994   Jan  1994.0   
2488  30-Jun-1993   Jun  1993.0   
2500  27-Mar-1993   Mar  1993.0   
2509  23-Jan-1993   Jan  1993.0   
2530  11-Oct-1992   Oct  1992.0   
2532  18-Sep-1992   Sep  1992.0   
2535  10-Sep-1992   Sep  1992.0   
2545  08-Jul-1992   Jul  1992.0   
2546  08-Jul-1992   Jul  1992.0   
2547  28-Jun-1992   Jun  1992.0   
2601  24-Apr-1991   Apr  1991.0   
3038     Mar-1981   Mar  1981.0   
3043     Feb-1981   Feb  1981.0   
3053  11-Nov-1980   Nov  1980.0   
3070  28-Mar-1980   Mar  1980.0   

                                               Location  \
6                                 Piedade Beach, Recife   
27                                              Piedade   
51                                      Conceicao Beach   
177                     Fernando de Noronha Marine Park   
181                                  Boa Viagem, Recife   
302                  Escalerio Beach Balneário Camboriú   
310                                  Estaleirinho beach   
324                                  Fernano de Noronha   
543                  500 km off the coast of Pernambuco   
651                            Boa Viagem Beach, Recife   
673                                    Coral Cove Beach   
751                 Coral Cove, Cabo de Santo Ago

# Analysis: Pernambuco cases in Brazil

In [80]:
attacks_brazil = attacks_br.groupby(by=['Year']).agg(Brazil = ('State', 'count'))

In [81]:
attacks_brazil

Brazil
Year          
1980.0       2
1981.0       2
1991.0       1
1992.0       6
1993.0       3
1994.0      10
1995.0       4
1996.0       4
1997.0       5
1998.0       4
1999.0       4
2001.0       2
2002.0       7
2003.0       5
2004.0       7
2005.0       1
2006.0       4
2007.0       1
2008.0       3
2009.0       2
2010.0       1
2011.0       2
2012.0       1
2013.0       2
2014.0       1
2015.0       1
2016.0       2
2017.0       2
2018.0       3

In [82]:
attacks_states = pd.crosstab(attacks_br['Year'],attacks_br['State'])

In [83]:
attacks_states

State   BA  MA  PE  RJ  RN  RS  SC
Year                              
1980.0   0   0   1   1   0   0   0
1981.0   0   0   0   2   0   0   0
1991.0   0   0   1   0   0   0   0
1992.0   0   3   3   0   0   0   0
1993.0   0   0   3   0   0   0   0
1994.0   0   0  10   0   0   0   0
1995.0   0   1   3   0   0   0   0
1996.0   0   0   4   0   0   0   0
1997.0   0   0   3   2   0   0   0
1998.0   0   0   4   0   0   0   0
1999.0   0   0   2   1   1   0   0
2001.0   0   0   1   0   1   0   0
2002.0   0   0   7   0   0   0   0
2003.0   0   0   2   3   0   0   0
2004.0   0   0   7   0   0   0   0
2005.0   1   0   0   0   0   0   0
2006.0   0   0   4   0   0   0   0
2007.0   0   0   1   0   0   0   0
2008.0   1   0   2   0   0   0   0
2009.0   0   0   2   0   0   0   0
2010.0   0   0   0   0   0   1   0
2011.0   0   0   2   0   0   0   0
2012.0   0   0   1   0   0   0   0
2013.0   0   0   2   0   0   0   0
2014.0   0   0   1   0   0   0   0
2015.0   0   0   1   0   0   0   0
2016.0   0   0   0   0   0   0   2
2017.0   0   0   2   0   0   0   0
2018.0   0   0   3   0   0   0   0

In [84]:
attacks_states = pd.merge(left=attacks_brazil, right=attacks_states, on='Year')

In [85]:
attacks_states

Brazil  BA  MA  PE  RJ  RN  RS  SC
Year                                      
1980.0       2   0   0   1   1   0   0   0
1981.0       2   0   0   0   2   0   0   0
1991.0       1   0   0   1   0   0   0   0
1992.0       6   0   3   3   0   0   0   0
1993.0       3   0   0   3   0   0   0   0
1994.0      10   0   0  10   0   0   0   0
1995.0       4   0   1   3   0   0   0   0
1996.0       4   0   0   4   0   0   0   0
1997.0       5   0   0   3   2   0   0   0
1998.0       4   0   0   4   0   0   0   0
1999.0       4   0   0   2   1   1   0   0
2001.0       2   0   0   1   0   1   0   0
2002.0       7   0   0   7   0   0   0   0
2003.0       5   0   0   2   3   0   0   0
2004.0       7   0   0   7   0   0   0   0
2005.0       1   1   0   0   0   0   0   0
2006.0       4   0   0   4   0   0   0   0
2007.0       1   0   0   1   0   0   0   0
2008.0       3   1   0   2   0   0   0   0
2009.0       2   0   0   2   0   0   0   0
2010.0       1   0   0   0   0   0   1   0
2011.0       2   0   0   2   0   0   0   0
2012.0       1   0   0   1   0   0   0   0
2013.0       2   0   0   2   0   0   0   0
2014.0       1   0   0   1   0   0   0   0
2015.0       1   0   0   1   0   0   0   0
2016.0       2   0   0   0   0   0   0   2
2017.0       2   0   0   2   0   0   0   0
2018.0       3   0   0   3   0   0   0   0